scorecard/product_pages.py

```diff
@@ -0,0 +1,98 @@
+import requests
+from .models import BusinessUnit, BusinessUnitGroup, Person, Product, ProductSecurityCapability, ProductSecurityRole, \
+    SecurityCapability, SecurityRole, Status
+from scorecard import scoring
+
+'''
+    Acquires all business unit and product data via the Product Pages API.
+'''
+
+PRODUCT_PAGES_API = 'https://pp.engineering.redhat.com/pp/api/latest'
+BUSINESS_UNITS_API = PRODUCT_PAGES_API + '/bugroups/'
+PRODUCTS_API = PRODUCT_PAGES_API + '/products/'
+SECURITY_CHAMPION = "Security Escalation Contact"
+PROGRAM_MANAGER = "Program Manager"
+PRODUCT_SECURITY = "Product Security"
+PRODUCT_SECURITY_PROGRAM_MANAGER = PRODUCT_SECURITY + " " + PROGRAM_MANAGER
+TRUSTED_CAS = '/etc/pki/ca-trust/extracted/openssl/ca-bundle.trust.crt'
+
+
+def update_product_data():
+
+    # Get all the business unit data and create or update accordingly.
+
+    bu_groups = requests.get(BUSINESS_UNITS_API, headers=dict(Accept='application/json'), verify=TRUSTED_CAS).json()
+    for bu_group in bu_groups:
+        updated_bu_group, _ = BusinessUnitGroup.objects.get_or_create(pp_id=bu_group['id'])
+        updated_bu_group.name = bu_group['name']
+        updated_bu_group.save()
+        business_units = bu_group['bus']
+        for bu in business_units:
+            updated_bu, _ = BusinessUnit.objects.get_or_create(pp_id=bu['id'])
+            updated_bu.name = bu['name']
+            updated_bu.bu_group = updated_bu_group
+            updated_bu.save()
+
+    # Get all the product data and create or update accordingly.
+    products = requests.get(PRODUCTS_API, headers=dict(Accept='application/json'), verify=TRUSTED_CAS).json()
```

Comment:
The ProductPages API also lets you restrict the response fields to only the ones you specify in a list. That can help it to respond much faster and be lighter on the required network traffic.